store/src/server/api.rs

```diff
@@ -1,39 +1,65 @@
-use std::net::ToSocketAddrs;
-use tracing::info;
-use anyhow::Result;
-
 use crate::config::StoreConfig;
-use tonic::{transport::Server, Response, Status, Streaming};
-
+use crate::db::Db;
+use anyhow::Result;
+use miden_crypto::merkle::TieredSmt;
+use miden_crypto::{Felt, FieldElement};
 use miden_node_proto::store;
+use tokio::time::Instant;
+use std::net::ToSocketAddrs;
+use tonic::{transport::Server, Response, Status, Streaming};
+use tracing::{instrument, info};
 
-#[derive(Default)]
-pub struct DBApi {}
+pub struct StoreApi {
+    db: Db,
+    nullifier_tree: TieredSmt,
+}
 
 #[tonic::async_trait]
-impl store::api_server::Api for DBApi {
+impl store::api_server::Api for StoreApi {
     type CheckNullifiersStream = Streaming<store::CheckNullifiersResponse>;
 
     async fn check_nullifiers(
         &self,
         _request: tonic::Request<Streaming<store::CheckNullifiersRequest>>,
-    ) -> Result<Response<Self::CheckNullifiersStream>, Status>
-    {
+    ) -> Result<Response<Self::CheckNullifiersStream>, Status> {
         todo!()
     }
 }
 
-pub async fn serve(config: StoreConfig) -> Result<()> {
+#[instrument(skip(db))]
```

Comment:
Question: what does `instrument(skip(db))` do?

Reply:
Bird's eye view of the feature is:

- `instrument` is a tracing macro that handles `async` code, (i.e. it tracks when the async task is entered/exited). And that is used by the subscribers (e.g. the logging library)
- `skip(db)` says to not log data of the argument `config` (disabled because it is too verbose)

Ref: https://docs.rs/tracing/latest/tracing/#spans-1